### DISCUSSION TITLE
Cloud && FastAPI: Auth Error[invalid_request]: g.

### DISCUSSION THREAD

**Discussion by michaldev** [fief-dev/fief#121] [Q&A]
## Describe the bug

I have an error that doesn't reveal any details.

## To Reproduce

1. Go to 127.0.0.1:8000/docs
2. Click on lock icon
3. Fill data and click authorize

![Zrzut ekranu z 2022-11-28 13-42-03](https://user-images.githubusercontent.com/5132385/204280466-8a8af792-8921-48a2-bfbb-c298a234c447.png)

**Comment by michaldev** [comment]
And on react:
![Zrzut ekranu z 2022-11-28 14-11-48](https://user-images.githubusercontent.com/5132385/204286167-6fe8a7f7-2731-4221-85e8-7f226d6c2cfd.png)

**Comment by frankie567** [comment]
Could you show the actual response of the request? You should be able to get it from the **Network** tab of your browser.

**Comment by michaldev** [comment]
`{error: "invalid_client"}`

I have correctly entered the client ID.

**Comment by frankie567** [comment]
Since you seem to call your client from a frontend like React, did you remember to set it as a [Public client](https://docs.fief.dev/admin-dashboard/clients/#public-clients)?

**Comment by michaldev** [comment]
Ahh, ok. I have confidental. But why this client doesn't work with fastapi?

**Comment by frankie567** [comment]
Maybe there's something else! If you show me the response of the request done by FastAPI auto documentation (again, by looking at the Network tab of your browser), I may have insights.

**Comment by michaldev** [comment]
In network I don't see anything. But I found something more. when I click on lock icon here (top position) it works:
![image](https://user-images.githubusercontent.com/5132385/204293032-89fa16ee-b990-4ecd-abac-79a32289855d.png)

Doesn't work when I click lock icon around the endpoint, like this:
![image](https://user-images.githubusercontent.com/5132385/204293299-45aa188f-b400-4d94-a982-9cd4b9c27f94.png)

**Comment by frankie567** [comment]
After investigating, I think I understand why it works with the top button, but not on the endpoint button: when you use the endpoint button, it doesn't ask you for scopes. However, Fief requires at least the `openid` scope, so that's probably the cause of the error.
